src/Control/Effect/NonDet.hs

```diff
@@ -1,11 +1,14 @@
-{-# LANGUAGE FlexibleInstances, LambdaCase, MultiParamTypeClasses, TypeOperators, UndecidableInstances #-}
+{-# LANGUAGE DeriveTraversable, FlexibleInstances, LambdaCase, MultiParamTypeClasses, TypeOperators, UndecidableInstances #-}
 module Control.Effect.NonDet
 ( NonDet(..)
 , Alternative(..)
 , runNonDet
 , AltC(..)
 , runNonDetOnce
 , OnceC(..)
+, Branch(..)
+, branch
+, runBranch
```

Comment:
I moved these here because they’re generally useful for implementing nondeterministic control effects.